examples/notion-github-sync/index.js

```diff
@@ -7,116 +7,230 @@
-  syncIssuesWithDatabase()
-})()
-
-// Get a paginated list of Tasks currently in a the database.
-async function getIssuesFromDatabase() {
-  const issues = {}
-
-  async function getPageOfIssues(cursor) {
-    let request_payload = ""
-    // Create the request payload based on the presence of a start_cursor.
-    if (cursor == undefined) {
-      request_payload = {
-        path: "databases/" + database_id + "/query",
-        method: "POST",
-      }
-    } else {
-      request_payload = {
-        path: "databases/" + database_id + "/query",
-        method: "POST",
-        body: {
-          start_cursor: cursor,
-        },
-      }
-    }
-    // While there are more pages left in the query, get pages from the database.
-    const current_pages = await notion.request(request_payload)
+/**
+ * Creates new pages in Notion.
+ *
+ * https://developers.notion.com/reference/post-page
+ *
+ * @param {Array<{ number: number, title: string, state: "open" | "closed", comment_count: number, url: string }>} pagesToCreate
+ */
+async function createPages(pagesToCreate) {
+  for (const pagesToCreateBatch of chunkItems(pagesToCreate)) {
+    await Promise.all(
+      pagesToCreateBatch.map(issue =>
+        notion.pages.create({
+          parent: { database_id: databaseId },
+          properties: getPropertiesFromIssue(issue),
+        })
+      )
+    )
+    console.log(`Completed batch size: ${pagesToCreateBatch.length}`)
+  }
+}
 
-    for (const page of current_pages.results) {
-      issues[page.properties["Issue Number"].number] = {
-        page_id: page.id,
-      }
-    }
-    if (current_pages.has_more) {
-      await getPageOfIssues(current_pages.next_cursor)
+/**
+ * Updates provided pages in Notion.
+ *
+ * https://developers.notion.com/reference/patch-page
+ *
+ * @param {Array<{ pageId: string, number: number, title: string, state: "open" | "closed", comment_count: number, url: string }>} pagesToUpdate
+ */
+async function updatePages(pagesToUpdate) {
+  for (const pagesToUpdateBatch of chunkItems(pagesToUpdate)) {
+    await Promise.all(
+      pagesToUpdateBatch.map(({ pageId, ...issue }) =>
+        notion.pages.update({
+          page_id: pageId,
+          properties: getPropertiesFromIssue(issue),
+        })
+      )
+    )
+    console.log(`Completed batch size: ${pagesToUpdateBatch.length}`)
+  }
+}
+
+//*========================================================================
+// Helpers
+//*========================================================================
+
+/**
+ * An iterator that batches out a list of items into smaller chunks.
+ */
+function* chunkItems(items) {
+  let startIndex = 0
+  while (true) {
+    const itemChunk = items.slice(startIndex, startIndex + OPERATION_BATCH_SIZE)
+    yield itemChunk
+    if (itemChunk.length < OPERATION_BATCH_SIZE) {
+      break
     }
+    startIndex += OPERATION_BATCH_SIZE
+  }
+}
+
+/**
+ * Returns the GitHub issue to conform to Notion database schema properties.
```

Comment:
still think "notion database schema properties" makes it sound like these are somehow canonical column names of the notion product, rather than of our particular DB

Reply:
instead of "Notion" i'll write "this database's schema properties" - these are things we use in our documentation
<img width="486" alt="image" src="https://user-images.githubusercontent.com/4068853/126692774-d6ccfee4-86d5-4e2c-985f-4e90d203ec4f.png">